src/RecoveryServices/RecoveryServices.Backup.Helpers/Validations/PolicyValidations.cs

```diff
@@ -246,21 +354,30 @@ private static void ValidateRetentionAndBackupTimes(List<DateTime> schPolicyTime
         }
 
         private static void ValidateDurationCountsForHourlyPolicy(LongTermRetentionPolicy ltrPolicy,
-            SimpleSchedulePolicy schPolicy)
+            SchedulePolicyBase schPolicyBase)
         {
-            if (ltrPolicy.DailySchedule != null && ltrPolicy.IsDailyScheduleEnabled == true)
+            if (schPolicyBase.GetType() == typeof(SimpleSchedulePolicy))
             {
-                if ( schPolicy != null && schPolicy.ScheduleRunFrequency == ScheduleRunType.Hourly)
-                {
-                    int numberOfPointsPerDay = (int)((schPolicy.ScheduleWindowDuration / schPolicy.ScheduleInterval) + 1);
-                    int totalNumberOfScheduledPoints = numberOfPointsPerDay * (ltrPolicy.DailySchedule.DurationCountInDays + 1); //Incorporating GC delays for Hourly schedules
+                SimpleSchedulePolicy schPolicy = (SimpleSchedulePolicy)schPolicyBase;
 
-                    if(totalNumberOfScheduledPoints > PolicyConstants.AfsDailyRetentionDaysMax)
+                if (ltrPolicy.DailySchedule != null && ltrPolicy.IsDailyScheduleEnabled == true && string.Compare(ltrPolicy.BackupManagementType, "AzureStorage") == 0)
+                {
+                    if (schPolicy != null && schPolicy.ScheduleRunFrequency == ScheduleRunType.Hourly)
                     {
-                        throw new ArgumentException(String.Format(Resources.DailyRetentionPointsLimitExceeded, PolicyConstants.AfsDailyRetentionDaysMax));
+                        int numberOfPointsPerDay = (int)((schPolicy.ScheduleWindowDuration / schPolicy.ScheduleInterval) + 1);
+                        int totalNumberOfScheduledPoints = numberOfPointsPerDay * (ltrPolicy.DailySchedule.DurationCountInDays + 1); //Incorporating GC delays for Hourly schedules
+
+                        if (totalNumberOfScheduledPoints > PolicyConstants.AfsDailyRetentionDaysMax)
+                        {
+                            throw new ArgumentException(String.Format(Resources.DailyRetentionPointsLimitExceeded, PolicyConstants.AfsDailyRetentionDaysMax));
+                        }
                     }
                 }
             }
+            else
+            {
+                // SimpleSchedulePolicyV2 hourly validations
```

Comment:
This code block is empty, is it expected?

Reply:
it was intentionality kept, but removing also fine. I removed it for now